### DISCUSSION TITLE
Generating CSV from Helm charts

### DISCUSSION THREAD

**Discussion by BlaineEXE** [rook/rook#9199] [Ideas]
I did the following to find CSV RBAC from generated files:
1. Run `make build`
2. Using the template directory from the build (in a temp file), cat out the CSV template RBAC:
 `cat /var/folders/ww/62wbqlm957z_m4blvbb34bbc0000gn/T/tmp.TfDeS3vm/cluster/olm/ceph/deploy/*.yaml`
3. Pipe that into `build/rbac/keep-rbac-yaml.py` to give a list of RBAC-related resources in the manifests.

[list-of-csv-rbac.txt](https://github.com/rook/rook/files/7557493/list-of-csv-rbac.txt)

I did the same thing as above for the non-generated `common.yaml`:
[list-of-common-rbac.txt](https://github.com/rook/rook/files/7557492/list-of-common-rbac.txt)

The diff between the two is below.
```
< ClusterRole/psp:rook
19d17
< ClusterRoleBinding/rook-ceph-object-bucket
22d19
< ClusterRoleBinding/rook-ceph-system-psp
27d23
< PodSecurityPolicy/00-rook-privileged
33d28
< Role/rook-ceph-purge-osd
39,40d33
< RoleBinding/rook-ceph-cmd-reporter-psp
< RoleBinding/rook-ceph-default-psp
42d34
< RoleBinding/rook-ceph-mgr-psp
45,46d36
< RoleBinding/rook-ceph-osd-psp
< RoleBinding/rook-ceph-purge-osd
51d40
< ServiceAccount/rook-ceph-purge-osd
```

From what I can tell, the CSV excludes the PSP and PSP-related resources. This makes an amount of sense since CSV will use SCC instead of PSP.

It also excludes newly-added rook-ceph-purge-osd resources, which I believe is a bug.

It also seems to ignore `ClusterRoleBinding/rook-ceph-object-bucket`, but I'm not sure why. There are `# OLM: {BEGIN,END} OBJECTBUCKET ROLEBINDING` comments for it, but it isn't picked up by the script currently. I think this is a bug if it's missing, but I don't have enough information to determine if this is a bug or not. @leseb please advise.

-----

I think we can generate the CSV from the helm chart in the same way that we hope to generate `common.yaml`. We can set `psp.enable=false` when doing the helm generation to do so pretty easily. We will get `rook-ceph-purge-osd` resources added, which I think we need. We will also get `ClusterRoleBinding/rook-ceph-object-bucket`, but we can filter that out if we need to.

For the final thing, I think it will be easiest to alter `cluster/olm/ceph/generate-rook-csv.sh` to remove the `OLM_INCLUDE_CEPHFS_CSI`, `OLM_INCLUDE_RBD_CSI`, and `OLM_INCLUDE_REPORTER` options that rely on the `# OLM` comments. @leseb do we ever use those options for generating upstream CSVs any more?

**Comment by BlaineEXE** [comment]
First draft of change to generate CSV from Helm here: https://github.com/rook/rook/pull/9200

**Reply by leseb** [reply]
Some answers to your questions, the `ServiceAccount/rook-ceph-purge-osd` not being present in the CSV is not an issue, since the primary consumer of it uses the `rook-ceph-system` SA, see: https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/job_templates.go#L267. Of course this is not ideal.

For the `ClusterRoleBinding/rook-ceph-object-bucket` I'm not sure why it's ignored.

For the CSI option, I don't remember from the top of my head if they are being used, but I suspect they will be needed on env where CephFS for instance is not enabled. So I'd suggest keeping this as is.
